web/packages/teleport/src/DeviceTrust/DeviceTrustLocked.tsx

```diff
@@ -0,0 +1,178 @@
+          userSelect: 'none',
+          height: '100px',
+        }}
+      >
+        <FakeDeviceList
+          items={generateFakeItems(15)}
+          fetchData={() => null}
+          fetchStatus={''}
+        />
+      </Box>
+      <StyledMessageContainer>
+        <Box bgColor={theme.colors.spotBackground[0]} p="3" borderRadius="50%">
+          <IconCircle Icon={LockIcon} size={64} />
+        </Box>
+        <Text fontSize="2" textAlign="center">
+          Device Trust enables trusted and authenticated device access. When
+          resources are configured with the Device Trust mode “required”,
+          Teleport will authenticate the Trusted Device, in addition to
+          establishing the user's identity and enforcing the necessary roles,
+          and leaves an audit trail with device information. For more details,
+          please view{' '}
+          <Link
+            href={
+              'https://goteleport.com/docs/access-controls/device-trust/guide/'
+            }
+            target="_blank"
+          >
+            Device Trust documentation
+          </Link>
+          .
+        </Text>
+        <Box>
+          <ButtonLockedFeature event={CtaEvent.CTA_TRUSTED_DEVICES}>
+            Unlock Device Trust with Teleport Enterprise
+          </ButtonLockedFeature>
+        </Box>
+      </StyledMessageContainer>
+    </FeatureBox>
+  );
+}
+
+const FakeDeviceList = ({
+  items = [],
+  pageSize = 20,
+  fetchStatus = '',
+  fetchData,
+}: DeviceListProps) => {
+  return (
+    <Table
+      data={items}
+      columns={[
+        {
+          key: 'osType',
+          headerText: 'OS Type',
+          render: ({ osType }) => <IconCell osType={osType} />,
+        },
+        {
+          key: 'assetTag',
+          headerText: 'Asset Tag',
+        },
+        {
+          key: 'enrollStatus',
+          headerText: 'Enroll Status',
+        },
+      ]}
+      emptyText="No Devices Found"
+      pagination={{ pageSize }}
+      fetching={{ onFetchMore: fetchData, fetchStatus }}
+    />
+  );
+};
+
+const IconCell = ({ osType }: { osType: TrustedDeviceOSType }) => {
+  let icon;
+  switch (osType) {
+    case 'Windows':
+      icon = <Windows size="small" mr={1} />;
+      break;
+    case 'Linux':
+      icon = <Linux size="small" mr={1} />;
+      break;
+    default:
+      icon = <Apple size="small" mr={1} />;
+  }
+  return (
+    <Cell align="left" style={{ display: 'flex' }}>
+      {icon} {osType}
+    </Cell>
+  );
+};
+
+const StyledMessageContainer = styled(Flex)`
+  position: relative;
+  top: 30%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  background-color: ${({ theme }) => theme.colors.levels.elevated};
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  padding: 24px;
+  gap: 24px;
+  width: 650px;
+  box-shadow: 0 5px 5px -3px rgba(0, 0, 0, 0.2),
+    0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12);
+  border-radius: 8px;
+`;
```

Comment:
Have we considered the possibility of moving these from `e/` to `shared/`, and then just injecting the fake data into this component, rather than having them duplicated?

Reply:
I gave it a thought but then decided to do it as its done currently because:

- The table required for `e` itself is holding up to another TODO (we want to use range based server side query but it is not currently implemented in Device Trust service, but can be updated in future). It still may not be a problem but then changes to the `e` table may require to update this one, which IMO brings unnecessary future work for the "fake list table". 
- We aren't sure if the design and CTA for this page will remain the same. The effort may not worth it if we want to update the design in future.

I would ideally move these into `shared/` only if we are to use actual functionality both in `oss` and `e`.